test/functional/groups/invites_controller_test.rb

```diff
@@ -65,4 +65,67 @@ def test_create_email_invite
     assert req.valid?
   end
 
+  def test_request_to_join_us_notifies_recipient
+    login_as @user
+    recipient = FactoryGirl.create(:user)
+
+    assert_difference 'RequestNotice.count' do
+      post :create, group_id: @group.to_param, recipients: recipient.name
+    end
+
+    notice = RequestNotice.last
+    assert_equal recipient.id, notice.user_id
+    assert_equal 'request_to_join_us', notice.data[:title]
+  end
+
+  def test_request_to_join_us_notifies_all_recipients
+    login_as @user
+    recipient1 = FactoryGirl.create(:user)
+    recipient2 = FactoryGirl.create(:user)
+
+    assert_difference('RequestNotice.count', 2) do
+      post :create, group_id: @group.to_param,
+        recipients: "#{recipient1.name}, #{recipient2.name}"
+    end
+
+    notices = RequestNotice.last(2)
+    assert_equal recipient1.id, notices.first.user_id
+    assert_equal recipient2.id, notices.last.user_id
+    assert_equal 'request_to_join_us', notices.first.data[:title]
+    assert_equal 'request_to_join_us', notices.last.data[:title]
+  end
+
+  def test_request_to_join_us_notifies_only_valid_recipients
+    login_as @user
+    recipient = FactoryGirl.create(:user)
+
+    assert_difference 'RequestNotice.count' do
+      post :create, group_id: @group.to_param,
+        recipients: "#{recipient.name}, #{@user.name}"
+    end
+
+    notice = RequestNotice.last
+    assert_equal recipient.id, notice.user_id
+    assert_equal 'request_to_join_us', notice.data[:title]
+  end
+
+  def test_invite_group_to_network_notifies_group_member
```

Comment:
I'm not sure this really makes sense. If I'm the member of a network and i invite your group to the network I think you should be notified so you can react upon it.
This is quite cumbersome though because groups may be large and they don't necessarily have a council. So  we might have to notify all the members of the group that was invited. What do you think, @aliaksandrb ?